File: a-tsioh_sandbox/gif_to_png.py
import sys
import os

from PIL import Image

def main(sources, target_dir):
    for s in sources:
        im = Image.open(s)
        im.convert("1").save(os.path.join(target_dir,os.path.basename(s).replace(".GIF",".png")))



if __name__ == "__main__":
    s_dir = sys.argv[1]
    t_dir = sys.argv[2]
    sources = [os.path.join(s_dir,p) for p in os.listdir(s_dir) if p.endswith(".GIF")]
    main(sources, t_dir)

File: scripts/font/gif-to-png.ts
#!/usr/bin/env bun
/**
 * Convert every `.GIF` in a directory to `.png` (RGBA) in another.
 * Port of a-tsioh_sandbox/gif_to_png.py, using `sharp` instead of Pillow.
 */
import { readdirSync } from "node:fs";
import { join } from "node:path";
import sharp from "sharp";

export async function gifToPng(sourceDir: string, targetDir: string): Promise<void> {
  const sources = readdirSync(sourceDir).filter((f) => f.endsWith(".GIF"));
  for (const name of sources) {
    const out = join(targetDir, name.replace(/\.GIF$/, ".png"));
    await sharp(join(sourceDir, name)).ensureAlpha().png().toFile(out);
  }
}

if (import.meta.main) {
  const [sourceDir, targetDir] = process.argv.slice(2);
  if (!sourceDir || !targetDir) {
    console.error("usage: bun run scripts/font/gif-to-png.ts <source-dir> <target-dir>");
    process.exit(1);
  }
  await gifToPng(sourceDir, targetDir);
}
